### PR TITLE
Add support for Xcode 8 and Swift 2.3

### PR DESCRIPTION
**Pull request by sharplet**

### CODE REVIEW

File: Rex.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -173,49 +173,49 @@
...
 			dstSubfolderSpec = 16;
 			files = (
-				D83457391AFEE4BE0070616A /* ReactiveCocoa.framework in (null) */,
-				D834573A1AFEE4BE0070616A /* Result.framework in (null) */,
+				D83457391AFEE4BE0070616A /* ReactiveCocoa.framework in CopyFiles */,
+				D834573A1AFEE4BE0070616A /* Result.framework in CopyFiles */,
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 		};
-		D8715DE81C21172A005F4191 /* (null) */ = {
```
Comment (NachoSoto) [inline Rex.xcodeproj/project.pbxproj:198]: LOL Xcode

**Comment by NachoSoto** [comment]
👍 
Can you update `.travis.yml` too?

File: Cartfile.resolved [hunk 2]
```diff
@@ -1,2 +1,2 @@
-github "antitypical/Result" "2.1.1"
-github "ReactiveCocoa/ReactiveCocoa" "v4.2.1"
+github "antitypical/Result" "2.1.3"
+github "ReactiveCocoa/ReactiveCocoa" "v4.2.2"
```
Comment (ikesyo) [inline Cartfile.resolved:2]: Explicitly requiring `~> 4.2.2` in `Cartfile` would be better I think.

Comment (sharplet) [inline Cartfile.resolved:2]: Good catch, done.

File: Rex.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -730,9 +730,11 @@
 					};
 					D83457131AFEE44E0070616A = {
 						CreatedOnToolsVersion = 6.3;
+						LastSwiftMigration = 0800;
```
Comment (ikesyo) [inline Rex.xcodeproj/project.pbxproj:735]: This should be added to all the targets in the project.

**Comment by ikesyo** [comment]
`pod lib lint` in `.travis.yml` would fail. Maybe we should add `--allow-warnings --quick`, thus the linter would not validate Xcode project integration.

File: Rex.xcodeproj/project.pbxproj (hunk 3 above)
Comment (sharplet) [inline Rex.xcodeproj/project.pbxproj:735]: Good point. Fixed.

**Comment by ikesyo** [comment]
[.travis.yml] `xcodebuild build-for-testing test-without-building` might be preferable to `xcodebuild test` on Xcode 8 (at least on Travis) as of https://github.com/Quick/Quick/pull/565/commits/ed9c17b8c39a5a36d0244a2e1ff4450a4b971a30.

**Comment by sharplet** [comment]
Ok, so it looks like CI passed on Xcode 8 but then failed with this:

```
/Users/travis/build.sh: line 109: shell_session_update: command not found
```

Seems like a Travis thing?

And the Xcode 7.3 build just never completes, timing out when building dependencies. Perhaps there should be a separate job for each Xcode/platform combination? That would be 8 total builds, but we'd gain concurrency and more granular failures across platforms.

**Comment by sharplet** [comment]
If someone's available, could https://travis-ci.org/RACCommunity/Rex/builds/151239865 be killed? It was misconfigured and none of the builds are starting (I've since pushed a fix).

**Comment by ikesyo** [comment]
I've just killed it :hocho:

**Comment by sharplet** [comment]
Ok, so the build _finally_ finished. The failures seem to be mostly be timeouts of some kind or another. I'm not sure what the best way to proceed is, the build times here are kinda ridiculous.

**Comment by andersio** [comment]
You may try `xcodebuild build-for-testing test-without-building` with iOS/tvOS, as mentioned by @ikesyo.

https://github.com/RACCommunity/Rex/pull/150#issuecomment-235119726
https://github.com/ReactiveCocoa/ReactiveCocoa/blob/master/.travis.yml

**Comment by c0diq** [comment]
Is there a plan to fix and merge this PR? I am currently blocked because I need this Swift 2.3 support for RAC 4.2.2 Thanks!!

**Comment by sharplet** [comment]
Ok, we're green! @c0diq hopefully we can get this in now.

**Comment by mdiep** [comment]
Looks good to me. 👍🏻

**Pull request merged**